subsys/prefetch/src/main/java/org/commonjava/indy/subsys/prefetch/PrefetchManager.java

```diff
@@ -54,42 +53,83 @@
     @Inject
     private PrefetchConfig config;
 
-    @WeftManaged
-    @ExecutorConfig( named = "prefetch-worker", threads = 5, priority = 1 )
     @Inject
-    private Executor executor;
+    private SpecialPathManager specialPathManager;
 
-    @Inject
-    private Instance<ContentListBuilder> listBuilders;
+    private volatile boolean stopped;
+
+    private final ExecutorService prefetchExecutor =
+            Executors.newFixedThreadPool( 5, new NamedThreadFactory( "Prefetch-Worker", true, 1 ) );
+
+    private final ExecutorService rescanScheduleExecutor =
+            Executors.newSingleThreadExecutor( new NamedThreadFactory( "Prefetch-Rescan-Scheduler", true, 1 ) );
+
+    private final Runnable rescanSchedulingTask = () -> {
```

Comment:
This looks like it might work better as a TimerTask.

Reply:
+1. Will change to use it.